File: app/src/main/java/com/pddstudio/highlightjs/demo/utils/FileObject.java
package com.pddstudio.highlightjs.demo.utils;

import java.io.File;
import java.io.Serializable;
import java.net.URL;

/**
 * This Class was created by Patrick J
 * on 13.06.16. For more Details and Licensing
 * have a look at the README.md
 */

public class FileObject implements Serializable {
    
    private final String name;
    private final URL url;

    FileObject(String name, URL url) {
        this.name = name;
        this.url = url;
    }

    public String getAbsoluteFilePath() {
        return name;
    }

    public URL getUrl() {
        return url;
    }

    public String getFileName() {
        return new File(name).getName();
    }

}

File: app/src/main/java/com/pddstudio/highlightjs/demo/utils/FileObject.kt
package com.pddstudio.highlightjs.demo.utils

import java.io.File
import java.io.Serializable
import java.net.URL

/**
 * This Class was created by Patrick J
 * on 13.06.16. For more Details and Licensing
 * have a look at the README.md
 */

class FileObject internal constructor(private val name: String, private val url: URL) : Serializable {

    fun getAbsoluteFilePath(): String = name

    fun getUrl(): URL = url

    fun getFileName(): String = File(name).name
}
